ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/mapper/ClaimantResponseMapperTest.java

```diff
@@ -31,82 +33,116 @@
+        assertNotNull(ccdResponse.getSubmittedOn());
     }
 
     @Test
-    public void shouldMapClaimantAcceptanceWithSettlementFormalisationFromCCD() {
-        //given
-        CCDClaimantResponse ccdResponse = SampleData.getResponseAcceptation(SETTLEMENT);
+    public void shouldMapClaimantAcceptanceWithCourtDeterminationPayByInstalmentsToCCDClaimantAcceptance() {
+        ClaimantResponse response = SampleClaimantResponse.ClaimantResponseAcceptation.builder()
+            .buildAcceptanceIssueSettlementWithCourtDeterminationPayByInstalments();
+        Claim claim =  claimBuilder.claimantResponse(response)
+            .claimantRespondedAt(LocalDateTimeFactory.nowInLocalZone())
+            .build();
+        CCDClaimantResponse ccdResponse = mapper.to(claim);
+        assertThat((ResponseAcceptation) response).isEqualTo((CCDResponseAcceptation)ccdResponse);
+        assertNotNull(ccdResponse.getSubmittedOn());
+    }
 
-        //when
-        ClaimantResponse response = mapper.from(ccdResponse).getClaimantResponse().get();
+    @Test
+    public void shouldMapClaimantAcceptanceWithClaimantPaymentIntentionPayImmediatelyToCCDClaimantAcceptance() {
+        ClaimantResponse response = SampleClaimantResponse.ClaimantResponseAcceptation.builder()
+            .buildAcceptanceIssueSettlementWithClaimantPaymentIntentionPayImmediately();
+        Claim claim =  claimBuilder.claimantResponse(response)
+            .claimantRespondedAt(LocalDateTimeFactory.nowInLocalZone())
+            .build();
+        CCDClaimantResponse ccdResponse = mapper.to(claim);
+        assertThat((ResponseAcceptation) response).isEqualTo((CCDResponseAcceptation)ccdResponse);
+        assertNotNull(ccdResponse.getSubmittedOn());
+    }
 
-        //then
-        assertThat((ResponseAcceptation)response).isEqualTo((CCDResponseAcceptation) ccdResponse);
+    @Test
+    public void shouldMapCCDResponseAcceptationWithCCJFormalisationToResponseAcceptation() {
+        CCDResponseAcceptation ccdResponse = SampleData.getResponseAcceptation(CCDFormaliseOption.CCJ);
+        mapper.from(ccdResponse,claimBuilder);
+        Claim claim = claimBuilder.build();
+        if (claim.getClaimantResponse().isPresent()) {
+            assertThat((ResponseAcceptation) claim.getClaimantResponse().get()).isEqualTo(ccdResponse);
+        }
+        claim.getClaimantRespondedAt().ifPresent(Assert::assertNotNull);
     }
 
     @Test
-    public void shouldMapClaimantRejectionFromCCD() {
-        //given
-        CCDResponseRejection ccdResponse = SampleData.getResponseRejection();
+    public void shouldMapCCDResponseAcceptationWithClaimantPaymentIntentionImmediatelyToResponseAcceptation() {
+        CCDResponseAcceptation ccdResponse = SampleData
+            .getResponseAcceptationWithClaimantPaymentIntentionImmediately();
+        mapper.from(ccdResponse,claimBuilder);
+        Claim claim = claimBuilder.build();
+        if (claim.getClaimantResponse().isPresent()) {
```

Comment:
this can be aligned with ifPresent

Reply:
Please use below instead of if statement in tests. I am not repeating this comment. Please do for all instances in tests.

```
assertThat(claim).isNotNull();
assertThat(claim.getClaimantResponse()).isPresent();
```

Reply:
Same as above. Please don't repeat this comment.